src/main/java/use_cases/add_edit_category_use_case/CategoryUCI.java

```diff
@@ -0,0 +1,143 @@
+package use_cases.add_edit_category_use_case;
+
+import entities.Category;
+import entities.EntityException;
+import entities.MonthlyStorage;
+
+import java.util.ArrayList;
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
+/**
+ * The CategoryUCI adds/creates new category, edits an existing category, and updates MonthlyStorage.categoryData.
+ * Implements CategoryIB.
+ * Creates CategoryOD objects related to specific use case fail/success conditions.
+ */
+
+public class CategoryUCI implements CategoryIB {
+    final CategoryOB categoryOB;
```

Comment:
This should be private final. It is standard to have everything in the most protected form that your code allows. So i would go back through every class to ensure that you have private final everywhere, because Ari commented on it too so it might be in other places.